plugins/jobtype/src/azkaban/jobtype/HadoopSecureSparkWrapper.java

```diff
@@ -210,22 +319,17 @@ private static void handleDynamicResourceAllocation(String[] argArray) {
     // feature for Yarn. This config inside Spark job type is to enforce node labeling feature for all
     // Spark applications submitted via Azkaban Spark job type.
     Configuration conf = new Configuration();
-    String sparkPropertyFile = HadoopSecureSparkWrapper.class.getClassLoader()
-        .getResource("spark-defaults.conf").getPath();
     boolean nodeLabelingYarn = conf.getBoolean(YARN_CONF_NODE_LABELING_ENABLED, false);
     String nodeLabelingProp = System.getenv(HadoopSparkJob.SPARK_NODE_LABELING_ENV_VAR);
     boolean nodeLabelingPolicy = nodeLabelingProp != null && nodeLabelingProp.equals(Boolean.TRUE.toString());
     String autoNodeLabelProp = System.getenv(HadoopSparkJob.SPARK_AUTO_NODE_LABELING_ENV_VAR);
     boolean autoNodeLabeling = autoNodeLabelProp != null && autoNodeLabelProp.equals(Boolean.TRUE.toString());
     String desiredNodeLabel = System.getenv(HadoopSparkJob.SPARK_DESIRED_NODE_LABEL_ENV_VAR);
-    String minMemVcoreRatio = System.getenv(HadoopSparkJob.SPARK_MIN_MEM_VCORE_RATIO_ENV_VAR);
-    String minMemGBSize = System.getenv(HadoopSparkJob.SPARK_MIN_MEM_SIZE_ENV_VAR);
     String executorMem = null;
     String executorVcore = null;
     String executorMemOverhead = null;
 
-    SparkConf sparkConf = new SparkConf(false);
-    sparkConf.setAll(Utils.getPropertiesFromFile(sparkPropertyFile));
+    SparkConf sparkConf = getSparkProperties();
 
     if (nodeLabelingYarn && nodeLabelingPolicy) {
       for (int i = 0; i < argArray.length; i++) {
```

Comment:
It seems that removing the queue parameter on line 350 in this method will interfere with the processing we are doing in handleQueueEnforcement.